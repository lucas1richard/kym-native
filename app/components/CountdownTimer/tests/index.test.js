import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { CountdownTimer } from '../';

describe('<CountdownTimer', () => {
  const renderedComponent = (props = {}) => shallow(<CountdownTimer {...props} />);

  it('should render without blowing up', () => {
    const props = {
      days: 3,
      hours: 14,
      minutes: 40,
      seconds: 40
    };
    expect(toJson(renderedComponent(props))).toMatchSnapshot();
  });

  it('should show minutes and seconds', () => {
    const props = {
      days: 0,
      hours: 0,
      minutes: 40,
      seconds: 40
    };
    expect(toJson(renderedComponent(props))).toMatchSnapshot();
  });
});
