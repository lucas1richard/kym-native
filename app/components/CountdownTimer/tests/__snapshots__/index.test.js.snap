// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountdownTimer should render without blowing up 1`] = `
<styled.div>
  <Styled(styled.span)>
    3
     days 
  </Styled(styled.span)>
  <Styled(styled.span)>
    14
     hours 
  </Styled(styled.span)>
</styled.div>
`;

exports[`<CountdownTimer should show minutes and seconds 1`] = `
<styled.div>
  <Styled(styled.span)>
    40
     minutes 
  </Styled(styled.span)>
  <Styled(styled.span)>
    40
     seconds 
  </Styled(styled.span)>
</styled.div>
`;
